src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -73,7 +74,7 @@ private static bool EndsWithPeriodOrSpace(string path)
         /// away from paths during normalization, but if we see such a path at this point it should be
         /// normalized and has retained the final characters. (Typically from one of the *Info classes)
         /// </summary>
-        internal static string EnsureExtendedPrefixIfNeeded(string path)
+        internal static string? EnsureExtendedPrefixIfNeeded(string? path)
```

Comment:
Should this have `[return: NotNullIfNotNull("path")]`?

Reply:
makes sense

Reply:
As talked offline was not able to add the attribute, for now leaving comment to add it later